megamek/src/megamek/server/Server.java

```diff
@@ -1022,23 +1022,21 @@ private IPlayer addNewPlayer(int connId, String name) {
             team++;
         }
         IPlayer newPlayer = new Player(connId, name);
-        int colorInd = newPlayer.getColorIndex();
+        PlayerColour colour = newPlayer.getColour();
         Enumeration<IPlayer> players = game.getPlayers();
-        while (players.hasMoreElements() && (colorInd < PlayerColors.COLOR_NAMES.length)) {
+        final PlayerColour[] colours = PlayerColour.values();
+        while (players.hasMoreElements()) {
             final IPlayer p = players.nextElement();
             if (p.getId() == newPlayer.getId()) {
                 continue;
             }
-            if (p.getColorIndex() == colorInd) {
-                colorInd++;
+            if ((p.getColour() == colour) && (colours.length < (colour.ordinal() + 1))) {
```

Comment:
Should this be equals?

Reply:
No, but it should be greater than instead of less than

That's there to prevent an index out of bounds error.